
# Profitable App Profiles for the App Store and Google Play Markets

This project aims to conduct an analysis of the company's app portfolio in order to determine which apps have the highest profitability. 

The objectives in this project are to determine the following:
-Which apps have the most users
-Which apps have the highest rating

The answers to these objectives will allow the data analysis team to determine which app should be targeted by advertisers.

In [2]:
#OpenGooglePlayStore

In [3]:
opened_file = open('googleplaystore.csv')
from csv import reader
read_file = reader(opened_file)
android = list(read_file)
androidheader = android[0]
android = android[1:]

In [4]:
#OpenAppleStore

In [5]:
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
apple = list(read_file)
appleheader = apple[0]
apple = apple[1:]

In [6]:
def explore_data(dataset, start, end, rows_and_columns = False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') # Adds a new (empty) line after each row
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [7]:
print(appleheader)
print('\n')
explore_data(apple, 0, 4, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7197
Number of columns: 16


In [8]:
print(androidheader)
print('\n')
explore_data(android, 0, 4, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10841
Number of columns: 13


# Column Descriptions

The following links contain information about the columns in both the Google and Apple Dataset, respectively. 

[Link]https://www.kaggle.com/lava18/google-play-store-apps

[Link]https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps

Android Useful Columns: App, Category, Reviews, Installs, Type, Price, Genres
Apple Useful Columns: track_name, currecy, price, rating_count_tot, rating_count_ver, prime_genre

In [9]:
print(android[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [10]:
del android[10472]

In [11]:
print(android[10472])

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


# Duplicate Entries
-The dataset contains duplicate entries in the GooglePlay (Android) Dataset. The code will now identify and remove the duplicate entries. 

In [12]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of Duplicate Entries:', len(duplicate_apps))
print('\n')
print('Examples of Duplicate Entries:', duplicate_apps[0:15])
print('\n')
print('Number of Unique Entries:', len(unique_apps))

Number of Duplicate Entries: 1181


Examples of Duplicate Entries: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


Number of Unique Entries: 9659


In [13]:
print(androidheader)

for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


# Duplicate Entry Removal

We noted that there are 1181 duplicate entries and 9659 unique entries. 

Now, we will attempt to remove the duplicate entries. The code will not remove duplicate entries at random. The code will seek to remove older entries and only maintain the most recent one. 

It will do this by checking which entry has the highest number of reviews, indicating that it is the most recent entry. The fourth index in each row corresponds to the number of reviews. 


The following step will create a for loop that will loop over all the apps in the android data set. We will start out by creating a blank dictionary. This blank dictionary will be appended everytime the loop encounters a new app not already contained in it. The append will add the name of the app and the number of reviews. If the loop encounters the same app with a greater number of reviews, it will replace the original number of reviews with the greater one.

In [14]:
reviews_max = {}
for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
print('Actual Length:', len(reviews_max))
        

Actual Length: 9659


The following step will create a new list using the information that we compiled in the dictionary previously. The for loop will check the number of reviews for every app across the list and check it against the keys and values stored in the dictionary. If the loop finds a match, it will add that entire row to the anroid_clean list. Simultaneously, we will keep track of each addition to android clean by adding the name to another list called already_added. This will help us keep track of which apps have been added already so we do not end up with duplicate entries once more, in the case of an two app entries that have the same number of reviews. 

In [15]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
print('Length of Android Clean:', len(android_clean))
print('Length of Already Added:', len(already_added))

Length of Android Clean: 9659
Length of Already Added: 9659


In [16]:
print(androidheader)
explore_data(android_clean, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


We will now define a function to check whether or not the app name is in English. The function will loop over the string (App Name) and check each character's corresponding number (using the ord function). English characters range from 0 - 127 according to the American Standard Code for Information Interchange (ASCII) system. If the character is greater than 127, the function will return false, meaning it is not in English. If it does not return false, it will return True, meaning the language is English. 

In [17]:
def languagecheck(input):
    for character in input:
        if ord(character) > 127:
            return False
    return True

In [18]:
languagecheck('爱奇艺PPS -《欢乐颂2》电视剧热播'
)

False

In [19]:
languagecheck('Instagram')

True

In [20]:
languagecheck('Docs To Go™ Free Office Suite')

False

In [21]:
languagecheck('Instachat 😜')

False

As we can see, the function does a relatively good job of checking the language. However, certain character symbols like emojis can lead to mislabeling in the function. We will now edit the function to only label input strings with more than three characters falling outside the ASCII range as non-english. 

In [22]:
def languagecheck(input):
    charactercheck = 0
    for character in input:
        if ord(character) > 127:
            charactercheck += 1
    if charactercheck > 3:
        return False
    else:
        return True

In [23]:
languagecheck('Instagram')

True

In [24]:
languagecheck('Docs To Go™ Free Office Suite')

True

In [25]:
languagecheck('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [26]:
languagecheck('Instachat 😜')

True

We will now create a new list from each data set (android_clean and apple) that only contain the apps in English.

In [27]:
Android_English = []
Apple_English = []

for app in android_clean:
    appname = app[0]
    if languagecheck(appname) == True:
        Android_English.append(app)
        
for app in apple:
    appname = app[1]
    if languagecheck(app[1]) == True:
        Apple_English.append(app)
        
explore_data(Android_English, 0, 3, True)
print('\n')
explore_data(Apple_English, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

Now we will isolate the free apps into a new dataset. Android_English price is in index 7 and Apple_English price is in index 4.

In [28]:
Android_Free = []
Apple_Free = []

for app in Android_English:
    appprice = app[7]
    if appprice == '0':
        Android_Free.append(app)
for app in Apple_English:
    appprice =app[4]
    if appprice == '0.0':
        Apple_Free.append(app)

print(len(Android_Free))
print(len(Apple_Free))

8864
3222


# Business Strategy
Our intent is to determine the kinds of apps that are likely to attact more users because our revenue is highly influenced by the number of people using our apps. To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps: 

1) Build a minimal Android version of the app, and add it to Google Play.
2) If the app has a good response from users, we develop it further
3) If the app is profitable after six months, we build an IOS version of the app and add it to the App store. 

We will want to find app profiles that are successful on both markets. Let's begin to analyze what are the most common genres for each market. Let's begin by building frequency tables. 

In [29]:
print(androidheader)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [30]:
print(appleheader)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


The most useful columns for our frequency table are likely to be category and genres for android. The most useful for apple would be prime_genre.

In [31]:
def freq_table(dataset, index):
    dataset_total = 0
    column_dictionary = {}
    for row in dataset:
        dataset_total += 1
        value = row[index]
        if value in column_dictionary:
            column_dictionary[value] += 1
        else:
            column_dictionary[value] = 1
    
    column_dictionary_percentages = {}
    for key in column_dictionary:
        percentage = (column_dictionary[key] / dataset_total) * 100
        column_dictionary_percentages[key] = percentage
    return column_dictionary_percentages

In [32]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

Display Frequency Table for Android Genres

In [37]:
display_table(Android_Free, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Display frequency table for Android Category

In [40]:
display_table(Android_Free, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

We can tell that the most common category on the Google Play store is "Family", with the runner up being "Game". In the genre column, the most cmmon genres are "tools", "entertainment", and "education". Drawing a pattern here is more difficult due to the partition between categories and genres. We will stick to categories for now since it is broader, comparable to Apple's Prime Genre column. The Google Play store seems to have a more balanced portfolio across the categories of apps. We will need to conduct a further analysis of ampunt of users per category to build our own app profile.

Display frequency table for Apple Prime Genre

In [41]:
display_table(Apple_Free, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Based on the frequency table we just generated for Apple's Prime Genre column, we can determine that the most common genre is "Games" with "Entertainment" being the runner up. It appears that most apps are designed for general entertainment purposes rather than practical lifestyle purposes. While this information is useful, we cannot recomend an app profile for the App Store market using this source alone. We will need to analyze the amount of users for that genre. 

In [42]:
freq_table(Apple_Free, 11)

{'Social Networking': 3.2898820608317814,
 'Photo & Video': 4.9658597144630665,
 'Games': 58.16263190564867,
 'Music': 2.0484171322160147,
 'Reference': 0.5586592178770949,
 'Health & Fitness': 2.0173805090006205,
 'Weather': 0.8690254500310366,
 'Utilities': 2.5139664804469275,
 'Travel': 1.2414649286157666,
 'Shopping': 2.60707635009311,
 'News': 1.3345747982619491,
 'Navigation': 0.186219739292365,
 'Lifestyle': 1.5828677839851024,
 'Entertainment': 7.883302296710118,
 'Food & Drink': 0.8069522036002483,
 'Sports': 2.1415270018621975,
 'Book': 0.4345127250155183,
 'Finance': 1.1173184357541899,
 'Education': 3.662321539416512,
 'Productivity': 1.7380509000620732,
 'Business': 0.5276225946617008,
 'Catalogs': 0.12414649286157665,
 'Medical': 0.186219739292365}

In [44]:
display_table(Apple_Free, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In [46]:
for genre in freq_table(Apple_Free, 11):
    total = 0
    len_genre = 0
    for row in Apple_Free:
        genre_app = row[11]
        if genre_app == genre:
            user_ratings = float(row[5])
            total += user_ratings
            len_genre += 1
    average_rating = total / len_genre
    print(genre, ':', average_rating)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


The Navigation genre of apps has the most users when using the total count of user ratings as a metric. The App Store does not have a total number of installs column like the Google Play Store. This workaround is not full proof but it does show us that Navigation apps have a high number of users and those users care significantly about the quality and effectiveness of the app. 

In [60]:
for category in freq_table(Android_Free, 1):
    total = 0
    len_category = 0
    for row in Android_Free:
        category_app = row[1]
        if category_app == category:
            number_installs = row[5]
            number_installs = number_installs.replace('+', '')
            number_installs = number_installs.replace(',','')
            number_installs = float(number_installs)
            total += number_installs
            len_category += 1
    average_number_installs = total / len_category
    print(category, ':', average_number_installs)


ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, communication apps have the most installations at 38,456,119. However, we want to try and find a category/genre that is popular on both the Apple and Google Play Stores. In this case, communication is popular in Android but non-existent in Apple. In the Apple store, the communication apps are likely included in "Social Networking". Let's check to see if this is the case. 

In [49]:
for app in Apple_Free:
    if app[11] == 'Social Networking':
        print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In [61]:
for app in Android_Free:
    if app[1] == 'COMMUNICATION':
        print(app[0],':', app[5])

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

In [63]:
for app in Android_Free:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                    or app[5] == '500,000,000+'
                                    or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

In [65]:
under_100_m = []

for app in Android_Free:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

If we remove the most popular apps in COMMUNICATION with over 100 million installs, the average decreases from 38,456,119 to 3,603,485. This is a significant decrease and shows us that a few apps skew the popularity of a category. Some of these apps include giants like Whatsapp, Gmail, and Hangouts. These giants are very difficult to compete with. So let's assess another category.

In [66]:
for app in Apple_Free:
    if app[11] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


The next popular category in the Apple Store is "Reference". Above, we can see what some of the Reference apps contain. We can see most of these apps are some kind of literature, dictionary, or translator. It appears this category is not oversaturated in the Apple market. Let's look at the equivalent category in Google Play store: 'Books and Reference'.


In [67]:
for app in Android_Free:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0],':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

We can see that this category is very similar, comprised of dictionaries, translators, and literature (particularly religious). Additionally, we can even see some tutorials on how to program.

In [70]:
for app in Android_Free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                    or app[5] == '500,000,000+'
                                    or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


We can see that only five apps in the Books and References category on Google Play store are undoubtedly popular. This means there is a good amount of market share to be captured here. 

Based on this analysis, we can see that the most popular apps are related to E-readers for books, book stores, or digitized religious scriptures. We can see that there are quite a few different apps for teaching programming. There appear to be some for programming in C and R. However, there are no apps for programming in Python. Developing an app for programming in Python can capture the initial market share of this sub-category and also attract users from existing programming apps in C and R. 

# Conclusion

In this project, we analyzed datasets from both the Apple App Store and the Google Play mobile app store with the intent of recommending an App profile that can be profitable in both stores. 

We concluded that creating an app for learning programming in python can be a profitable course of action. The app can capture existing literature from python courses and books while also incorporating quizzes, daily tips, and discussion forums. 